### DISCUSSION TITLE
Using translations with IO's Render Framework

### DISCUSSION THREAD

**Discussion by matheusps** [vtex/shoreline#446] [Blog]
In the Design System, some components have default translations. To switch locales you can use the [`I18nProvider` component](https://admin-ui.vercel.app/experimental/localization#i18nprovider), exported by our package. To get the locale from the Render framework, you can use `useRuntime` like this:

```tsx
import { useRuntime } from 'vtex.render-runtime'
import { I18nProvider } from '@vtex/admin-ui'

export function Example() {
  const { culture: { locale } } = useRuntime()
  
  return (
    <I18nProvider locale={locale}>
      {/** your app goes here */}
    </I18nProvider>
  )
}
```

If you often use the locale, you have the option of abstracting the behavior on a custom hook:

```tsx
import { useRuntime } from 'vtex.render-runtime'

export function useLocale(): string {
  const { culture: { locale } } = useRuntime()

  return locale
}
```

**Comment by davicostalf** [comment]
@matheusps Can this discussion be merged in #579?

**Reply by matheusps** [reply]
This is outaded
